api-service/src/controllers/CreateQueryTemplate/CreateTemplateController.ts

```diff
@@ -0,0 +1,75 @@
+import { Request, Response } from "express";
+import logger from "../../logger";
+import { schemaValidation } from "../../services/ValidationService";
+import validationSchema from "./CreateTemplateValidationSchema.json";
+import { ResponseHandler } from "../../helpers/ResponseHandler";
+import { getQueryTemplate } from "../../services/QueryTemplateService";
+import * as _ from "lodash";
+import { validateTemplate } from "./QueryTemplateValidator";
+import { QueryTemplate } from "../../models/QueryTemplate";
+import slug from "slug";
+const apiId = "query.template.create";
+
+export const createQueryTemplate = async (req: Request, res: Response) => {
+    try {
+        const isValidName = validateName(req.params?.templateName);
+        if (!isValidName) {
+            logger.error({ apiId, message: "Invalid name provided", code: "QUERY_TEMPLATE_INVALID_INPUT" })
+            return ResponseHandler.errorResponse({ message: "Template name should contain alphanumeric characters and single space between characters", statusCode: 400, errCode: "BAD_REQUEST", code: "QUERY_TEMPLATE_INVALID_INPUT" }, req, res);
+        }
+        const templateName: string = req.params?.templateName;
+        const templateId: string = slug(templateName, '-');
+        const requestBody = req.body;
+        const isValidSchema = schemaValidation(requestBody, validationSchema)
+
+        if (!isValidSchema?.isValid) {
+            logger.error({ apiId, message: isValidSchema?.message, code: "QUERY_TEMPLATE_INVALID_INPUT" })
```

Comment:
add request.msgid and resmsgid in each logs as discussed

Reply:
done